### ISSUE TITLE
'Covariance of type variables used in super-interfaces is now enforced'

### ISSUE THREAD

**Issue by modulovalue** [spebbe/dartz#12]
Bad news: [CHANGELOG](https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md)

Leads to:

```
Compiler message:
file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/src/endo.dart:5:7: Error: Found unsupported uses of 'A' in supertype 'Monoid'.
class EndoMonoid<A> extends Monoid<Endo<A>> {
      ^
file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/src/endo.dart:11:33: Error: A value of type 'EndoMonoid<dynamic>' can't be assigned to a variable of type 'Monoid<dynamic Function(dynamic)>'.
 - 'EndoMonoid' is from 'package:dartz/dartz.dart' ('file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/dartz.dart').
 - 'Monoid' is from 'package:dartz/dartz.dart' ('file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/dartz.dart').
Try changing the type of the left hand side, or casting the right hand side to 'Monoid<dynamic Function(dynamic)>'.
final Monoid<Endo> EndoMi = new EndoMonoid();
                                ^
file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/src/endo.dart:12:36: Error: A value of type 'EndoMonoid<dynamic>' can't be assigned to a variable of type 'Monoid<A Function(A)>'.
 - 'EndoMonoid' is from 'package:dartz/dartz.dart' ('file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/dartz.dart').
 - 'Monoid' is from 'package:dartz/dartz.dart' ('file:///Users/-/.pub-cache/git/dartz-c793e317bfdca1062144f33e4cb224704435ebe6/lib/dartz.dart').
Try changing the type of the left hand side, or casting the right hand side to 'Monoid<A Function(A)>'.
Monoid<Endo<A>> endoMi<A>() => new EndoMonoid();
                                   ^
```

**Comment by spebbe** [comment]
Awww, man...

Thanks for reporting and for your patch! I tried another method, as well. Killing `EndoMonoid` and redefining `EndoMi` `endoMi` in terms of `_AnonymousMonoid` seems to fix things with fewer changes and without having to box endomorphisms:
```dart
Monoid<Endo<A>> endoMi<A>() => monoid(() => (A a) => a, (Endo<A> e1, Endo<A> e2) => (A a) => e1(e2(a)));
final Monoid<Endo> EndoMi = endoMi();
```

Thoughts?

**Comment by modulovalue** [comment]
That's definitely better than my naive fix and thanks for the quick response!

**Comment by spebbe** [comment]
Great! I'll commit and publish a release soon!

**Comment by spebbe** [comment]
Released in 0.8.5 from master and rebased onto 0_9_0_wip_2
